src/Microdown-DocumentBrowser/MicPharoPackageCommentResourceReference.class.st

```diff
@@ -0,0 +1,208 @@
+"
+My prime purpose it to allow access to the package comment.
+
+My uri syntax is: `comment://package/categoryName`.
+
+## loadMicrodown
+My `loadMicrodown` returns the package comment of the package named categoryName or empty text if categoryName is not a package or if the package has no comment. Notice, `loadMicrodown` will return the synthesized comment (created by the class side method `buildMicroDownUsing:withComment:`)
+
+When categoryName is a package, my two methods `contents` and `contents:` reads and writes the contents of the package comment.
+
+## loadDirectory
+My 'loadDirectory' has been designed to provide a deep directory. 
+
+Most class category names are of the form 'aaa-bbb-ccc-ddd'. A call to `loadDirectory` with categoryName being 'aaa' will give children 'aaa-xxx','aaa-yyy','aaa-zzz'. A call with 'aaa-bbb' will return children 'aaa-bbb-xxx', 'aaa-bbb-yyy' etc.
+
+### Details
+In `comment://package/categoryName`, categoryName can be one of three cases:
+* A package name, with children being the tags or classes if the package has no tags
+* A tag of the form 'PackageName-tag', with children being the classes of the tag
+* A package name prefix of the form 'AI-Algorithms' wich is neither a package or a tag in a package, with children being those categories of which 'AI-Algorithms' is a prefix.
+
+A special case is somewhat common, for example 'Microdown' is both a package name and a prefix for other packages.
+In this case, the prefix 'Microdown' takes precedence  over package, and `comment://package/Microdown` has as children those categories prefixed my 'Microdown' _and_ the Microdown package.
+
+"
+Class {
+	#name : #MicPharoPackageCommentResourceReference,
+	#superclass : #MicPharoCommentResourceReference,
+	#instVars : [
+		'kind',
+		'categoryName'
+	],
+	#category : #'Microdown-DocumentBrowser-ResourceModel'
+}
+
+{ #category : #private }
+MicPharoPackageCommentResourceReference >> browserIcon [
+	self kind = #prefix ifTrue: [ ^ self iconNamed: #packageColored ].
+	self kind = #package ifTrue: [ ^ self iconNamed: #package ].
+	self kind = #tag ifTrue: [ ^ self iconNamed: #book ].
+	^ self iconNamed: #error
+]
+
+{ #category : #testing }
+MicPharoPackageCommentResourceReference >> canSave [
+	"return true if I implement contents: "
+	^self isPackage
+]
+
+{ #category : #accessing }
+MicPharoPackageCommentResourceReference >> categoryName [
+
+	^ categoryName
+]
+
+{ #category : #private }
+MicPharoPackageCommentResourceReference >> childrenOfAllCategories [
+	"I return the children of 'comment://package/' "
+	^ (SystemOrganizer default categories 
```

Comment:
pay attention that this categories stuff should disappear in the future. 
It will take some years so we have time.

Reply:
OK, I will wait :-)
The reason I put it in is simply to get some deeper level of nesting, not the least to support that all microdown projects are grouped together.